crates/bevy_solari/src/scene/binder.rs

```diff
@@ -0,0 +1,442 @@
+
+        if material.emissive != LinearRgba::BLACK {
+            light_sources
+                .get_mut()
+                .push(GpuLightSource::new_emissive_mesh_light(
+                    instance_id as u32,
+                    (index_slice.range.len() / 3) as u32,
+                ));
+        }
+
+        instance_id += 1;
+    }
+
+    if instance_id == 0 {
+        return;
+    }
+
+    for directional_light in &directional_lights_query {
+        let directional_lights = directional_lights.get_mut();
+        let directional_light_id = directional_lights.len() as u32;
+
+        directional_lights.push(GpuDirectionalLight {
+            direction_to_light: directional_light.transform.back().into(),
+            cos_theta_max: (SUN_ANGULAR_DIAMETER_RADIANS / 2.0).cos(),
+            illuminance: directional_light.color * directional_light.illuminance,
+        });
+
+        light_sources
+            .get_mut()
+            .push(GpuLightSource::new_directional_light(directional_light_id));
+    }
+
+    materials.write_buffer(&render_device, &render_queue);
+    transforms.write_buffer(&render_device, &render_queue);
+    geometry_ids.write_buffer(&render_device, &render_queue);
+    material_ids.write_buffer(&render_device, &render_queue);
+    light_sources.write_buffer(&render_device, &render_queue);
+    directional_lights.write_buffer(&render_device, &render_queue);
+
+    let mut command_encoder = render_device.create_command_encoder(&CommandEncoderDescriptor {
+        label: Some("build_tlas_command_encoder"),
+    });
+    command_encoder.build_acceleration_structures(&[], [&tlas]);
+    render_queue.submit([command_encoder.finish()]);
+
+    raytracing_scene_bindings.bind_group = Some(render_device.create_bind_group(
+        "raytracing_scene_bind_group",
+        &raytracing_scene_bindings.bind_group_layout,
+        &BindGroupEntries::sequential((
+            vertex_buffers.as_slice(),
+            index_buffers.as_slice(),
+            textures.as_slice(),
+            samplers.as_slice(),
+            materials.binding().unwrap(),
+            tlas.as_binding(),
+            transforms.binding().unwrap(),
+            geometry_ids.binding().unwrap(),
+            material_ids.binding().unwrap(),
+            light_sources.binding().unwrap(),
+            directional_lights.binding().unwrap(),
+        )),
+    ));
+}
+
+impl FromWorld for RaytracingSceneBindings {
+    fn from_world(world: &mut World) -> Self {
+        let render_device = world.resource::<RenderDevice>();
+
+        let bind_group_layout_entries = &[
+            BindGroupLayoutEntry {
+                binding: 0,
+                visibility: ShaderStages::COMPUTE,
+                ty: BindingType::Buffer {
+                    ty: BufferBindingType::Storage { read_only: true },
+                    has_dynamic_offset: false,
+                    min_binding_size: None,
+                },
+                count: MAX_MESH_COUNT,
+            },
```

Comment:
I thought there was a shorthand api for this

Reply:
Not for entries with count, idt. If someone knows otherwise, let me know.